src/components/i-calendar.vue

```diff
@@ -0,0 +1,537 @@
+  },
+  data() {
+    return {
+      activeDate: undefined,
+      activeMonth: null,
+      activeYear: null,
+      selectedDate: [],
+      activeMonthPicker: undefined,
+      activeYearPicker: undefined,
+      showMonthYearTable: false,
+    };
+  },
+  computed: {
+    listDays() {
+      return Array.from(Array(7), (v, i) =>
+        dayjs()
+          .day(i + 0)
+          .format('ddd'),
+      );
+    },
+    listMonth() {
+      const x = Array.from(Array(12), (v, i) => ({
+        monthDisplay: dayjs(this.activeYearPicker)
+          .month(i + 0)
+          .format('MMM'),
+        monthValue: dayjs(this.activeYearPicker)
+          .month(i + 0)
+          .startOf('month'),
+      }));
+
+      return x;
+    },
+    activeDateString() {
+      return this.activeDate ? this.activeDate.toString() : undefined;
+    },
+    activeYearPickerString() {
+      return dayjs(this.activeYearPicker.toString()).format('YYYY');
+    },
+    previousPicker() {
+      const display = [];
+
+      const previous = dayjs(this.activeDateString).date(0);
+      const current = dayjs(this.activeDateString).date(0);
+      for (let i = 0; i <= current.day(); i += 1) {
+        display.push(previous.subtract(i, 'day'));
+      }
+      return display.sort((a, b) => a.day() - b.day());
+    },
+    currentPicker() {
+      return Array.from(Array(dayjs(this.activeDateString).daysInMonth()), (v, i) => {
+        const date = dayjs(this.activeDateString).date(++i);
+        return { date };
+      });
+    },
+    nextPicker() {
+      const display = [];
+      const previous = this.previousPicker.length;
+      const current = dayjs(this.activeDateString).daysInMonth();
+      for (let i = 1; i <= 42 - (previous + current); i += 1) {
+        display.push(dayjs(this.activeDateString).date(i).add(1, 'month'));
+      }
+      return display;
+    },
+  },
+  watch: {
+    activeDate: {
+      immediate: true,
+      handler(val) {
+        if (val) {
+          this.activeMonth = dayjs(val.toString()).format('MMM');
+          this.activeYear = dayjs(val.toString()).format('YYYY');
+        }
+      },
+    },
+    value: {
+      handler(val) {
+        const valueArray = Array.isArray(val) ? val : [val];
+
+        const isValueSame = this.checkSame(valueArray, this.selectedDate);
+        if (!isValueSame) {
+          this.selectedDate = valueArray.filter((date) => !!date).map((date) => dayjs(date.toString()).toDate());
+        }
+      },
+    },
+  },
+  created() {
+    this.activeDate = this.initialDate;
+  },
+  mounted() {
+    if (this.value) {
+      if (Array.isArray(this.value)) {
+        if (this.value.length > 0) {
+          const [first] = this.value;
+          this.activeDate = dayjs(first.toString()).toDate();
+          const results = this.value.map((date) => dayjs(date.toString()).toDate());
+          this.selectedDate = results;
+        } else {
+          this.activeDate = dayjs().toDate();
+        }
+      } else {
```

Comment:
you can remove all function inside if (Array.isArray(this.value)) {}
** BUT don't forget to remove array type in 'value' props